snap/info.go

```diff
@@ -254,12 +274,16 @@ func ReadInfo(name string, si *SideInfo) (*Info, error) {
 		return nil, err
 	}
 
-	info, err := InfoFromSnapYaml(meta)
+	return infoFromSnapYamlWithSideInfo(meta, si, false)
```

Comment:
Tiny nitpick: what the boolean parameter means is only clear if you look at the function declaration. I wonder if we could change ReadInfoFromSnapFile to to validate as an extra step (seems to be the only place that validates). Or alternatively use a flag so that we can write something like: `infoFromSnapYamlWithSideInfo(meta, si, NoValidate)`.

Not a blocker though.